test/suites/cloud-init.sh

```diff
@@ -27,4 +27,68 @@ test_cloud_init() {
   [ -n "${ID6}" ] && [ "${ID6}" != "${ID5}" ]
 
   lxc delete -f c1 c2
+
+  fingerprint="$(lxc image info testimage | awk '/^Fingerprint:/ {print $2}')"
+
+  # Ensure testimage is not set as cached.
+  lxd sql global "UPDATE images SET cached=0 WHERE fingerprint=\"${fingerprint}\""
+
+  (
+    cd devlxd-client || return
+    # Use -buildvcs=false here to prevent git complaining about untrusted directory when tests are run as root.
+    CGO_ENABLED=0 go build -tags netgo -v -buildvcs=false ./...
+  )
+
+  lxc launch testimage devlxd
+  lxc file push --quiet --mode 0755 "devlxd-client/devlxd-client" devlxd/bin/
+
+  # Unknown cloud-init formats should be passed to the instance unmodified
+  lxc config set devlxd cloud-init.user-data="invalid-yaml"
+  [ "$(lxc exec devlxd -- devlxd-client cloud-init user-data)" = "invalid-yaml" ]
+  lxc config unset devlxd cloud-init.user-data
+
+  # With SSH keys configured, unknown cloud-init formats should be passed to the instance unmodified
+  lxc config set devlxd cloud-init.user-data="invalid-yaml"
+  lxc config set devlxd cloud-init.ssh-keys.mykey="root:gh:user1"
+  [ "$(lxc exec devlxd -- devlxd-client cloud-init user-data)" = "invalid-yaml" ]
+  lxc config unset devlxd cloud-init.ssh-keys.mykey
+  lxc config unset devlxd cloud-init.user-data
+
+  # Configured ssh-keys in user-data should not replace vendor-data with users provided
+  CLOUD_INIT="#cloud-config
+users:
+  - name: root
+    ssh-import-id: gh:user2"
+
+  lxc config set devlxd cloud-init.vendor-data="${CLOUD_INIT}"
+  lxc config set devlxd cloud-init.ssh-keys.mykey="root:gh:user1"
+  lxc exec devlxd -- devlxd-client cloud-init vendor-data | grep "gh:user1"
+  lxc exec devlxd -- devlxd-client cloud-init vendor-data | grep "gh:user2"
+  [ "$(lxc exec devlxd -- devlxd-client cloud-init user-data)" = "not found" ]
```

Comment:
Should we expect a non-0 return from `devlxd-client` or is that client just very simplistic/only for testing purposes?